topics/Routing_in_Ktor.md

```diff
@@ -122,6 +124,67 @@ To access a parameter value inside the route handler, use the `call.parameters`
 A path parameter with a tailcard (`{param...}`) matches all the rest of the URL path and puts multiple values for each path segment into parameters using `param` as a key. For example, `/user/{param...}` matches `/user/john/settings`.
 To access path segments' values inside the route handler, use `call.parameters.getAll("param")`. For the example above, the `getAll` function will return an array containing the _john_ and _settings_ values.
 
+### Regular expression {id="regular_expression"}
+
+Regular expressions can be used with all defining route handlers functions: `route`, `get`, `post`, and so on. 
+
+> To know more about regular expressions, see [Kotlin documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/-regex/).
+
+Let's write a route matches any path that ends with `/hello`.
+
+```kotlin
+import io.ktor.server.routing.*
+import io.ktor.server.response.*
+
+routing {
+    get(Regex(".+/hello")) {
+        call.respondText("Hello")
+    }
+}
+```
+With this route definition, any incoming request to a path ending with `/hello`, such as `/foo/hello`, `/bar/baz/hello`, 
+and so on, will be matched.
+
+#### Accessing path parts in handler
+
+In regular expressions, named groups are a way to capture a specific part of a string that matches a pattern and assign it a name.
+Named groups are defined using the following syntax `(?<name>pattern)`, where `name` is the name of the group and
+`pattern` is the regular expression pattern that matches the group.
+
+By defining a named group in the route function, you can capture a part of the path, and then in the handler function,
+you can access the captured parameter using the `call.parameters` property.
+
+For example, you can define a route that matches requests to a path that includes an integer identifier followed by `/hello`.
+
+```kotlin
+import io.ktor.server.routing.*
+import io.ktor.server.response.*
+
+routing {
+    get(Regex("(?<id>\\d+)/hello")) {
+        val id = call.parameters["id"]!!
+        call.respondText(id)
+    }
+}
+```
+In the code below, the `(?<id>\\d+)` named group is used to capture the integer identifier `id` from the path,
+and the `call.parameters` property is used to access the captured id parameter in the handler function.
+
+Unnamed groups can't be accessed inside the route handler, but you can use them to match the path. For example, path `hello/world` will be matched while `hello/World` not:
```

Comment:
Unnamed groups can't be accessed inside **a regex route handler**
There are no unnamed groups in the following example.